test/Microsoft.AspNetCore.Authentication.Test/CookieTests.cs

```diff
@@ -900,6 +900,76 @@ private Task SignOutAsWrong(HttpContext context)
             Assert.Null(FindClaimValue(transaction5, ClaimTypes.Name));
         }
 
+        [Fact]
+        public async Task CookieCanBeReplacedByValidator()
+        {
+            var server = CreateServer(o =>
+            {
+                o.Events = new CookieAuthenticationEvents
+                {
+                    OnValidatePrincipal = ctx =>
+                    {
+                        ctx.ShouldRenew = true;
+                        ctx.ReplacePrincipal(new ClaimsPrincipal(new ClaimsIdentity(new GenericIdentity("Alice2", "Cookies2"))));
```

Comment:
Is it not possible to update the AuthProperties at the same time?

Reply:
Hrm, we might have a similar bug with properties I guess due to the cloning change.  Renew probably is stomping on any changes that were applied in the context

Reply:
There's no issue with properties because the context ends up taking the ticket properties 

https://github.com/aspnet/Security/blob/master/src/Microsoft.AspNetCore.Authentication.Cookies/Events/CookieValidatePrincipalContext.cs#L23

so updates to the ticket's properties show up automatically when they get copied:

https://github.com/aspnet/Security/blob/master/src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs#L113

since ticket.Properties == context.Properties

I'll update the test to set a property, but I don't think we need to change the fix